controllers/client_policy.go

```diff
@@ -169,18 +171,36 @@ func (r *SingleClusterReconciler) getClusterServerCAPool(
 	}
 }
 
-func (r *SingleClusterReconciler) appendCACertFromFile(
+func (r *SingleClusterReconciler) appendCACertFromFileOrPath(
 	caPath string, serverPool *x509.CertPool,
 ) *x509.CertPool {
 	if caPath == "" {
 		r.Log.Info("CA path is not provided in \"operatorClientCertSpec\". Using default system CA certs...")
-	} else if caData, err := os.ReadFile(caPath); err != nil {
+		return serverPool
+	}
+
+	err := filepath.WalkDir(
```

Comment:
ca-cert path can be file and dir both. Till now ca-cert path was used for the file path.

Reply:
yes, even if caPath is a file or dir, this function will handle both.